utils/src/deployer.rs

```diff
@@ -113,565 +119,603 @@ impl Contracts {
-
-        let (genesis_lc, genesis_stake) = genesis.await?.clone();
-
-        let data = light_client
-            .initialize(genesis_lc.into(), genesis_stake.into(), 864000, deployer)
-            .calldata()
-            .context("calldata for initialize transaction not available")?;
-        let light_client_proxy_address = contracts
-            .deploy_tx(
-                Contract::LightClientProxy,
-                ERC1967Proxy::deploy(l1.clone(), (lc_address, data))?,
-            )
-            .await?;
-
-        // confirm that the implementation address is the address of the light client contract deployed above
-        if !is_proxy_contract(&provider, light_client_proxy_address)
-            .await
-            .expect("Failed to determine if light contract is a proxy")
-        {
-            panic!("Light Client contract's address is not a proxy");
-        }
-
-        // Instantiate a wrapper with the proxy address and light client ABI.
-        let proxy = LightClient::new(light_client_proxy_address, l1.clone());
-
-        // Perission the light client prover.
-        if let Some(prover) = permissioned_prover {
-            tracing::info!(%light_client_proxy_address, %prover, "setting permissioned prover");
-            proxy.set_permissioned_prover(prover).send().await?.await?;
-        }
-
-        // Transfer ownership to the multisig wallet if provided.
-        if let Some(owner) = multisig_address {
-            tracing::info!(
-                ?light_client_proxy_address,
-                ?owner,
-                "transferring light client proxy ownership to multisig",
-            );
-            proxy.transfer_ownership(owner).send().await?.await?;
-        }
+    // sanity check
+    if !is_proxy_contract(&provider, lc_proxy_addr).await? {
+        panic!("LightClientProxy detected not as a proxy, report error!");
     }
 
-    // `FeeContract.sol`
-    if should_deploy(ContractGroup::FeeContract, &only) {
-        let fee_contract_address = contracts
-            .deploy_tx_alloy(
-                Contract::FeeContract,
-                FeeContract::deploy_builder(l1_alloy.clone()),
-            )
-            .await?;
-        let fee_contract = FeeContract::new(fee_contract_address, l1_alloy.clone());
-        let data = fee_contract
-            .initialize(deployer.to_alloy())
-            .calldata()
-            .clone();
-        let fee_contract_proxy_address = contracts
-            .deploy_tx(
-                Contract::FeeContractProxy,
-                ERC1967Proxy::deploy(
-                    l1.clone(),
-                    (fee_contract.address().to_ethers(), data.to_ethers()),
-                )?,
-            )
+    // instantiate a proxy instance, cast as LightClient's ABI interface
+    let lc_proxy = LightClient::new(lc_proxy_addr, &provider);
+
+    // set permissioned prover
+    if let Some(prover) = prover {
+        tracing::info!(%lc_proxy_addr, %prover, "Set permissioned prover ");
```

Comment:
@alxiong in the native demo this fails with `    /**Custom error with signature `OwnableUnauthorizedAccount(address)` and selector `0x118cdaa7`.`

So I think if we set another admin we need to do it after we set the permissioned prover.

Reply:
> if we set another admin we need to do it after we set the permissioned prover.

But that's what I do atm, see: https://github.com/EspressoSystems/espresso-network/blob/79c95ba8382ced91553484167edb86ad92c72fd9/sequencer/src/bin/deploy.rs#L223-L237

I set the permissioned prover with deployer (inside `deployer::deploy_light_client_proxy()`), then I did the `transfer_ownership(multisig)`. 
^^ this should be correct, right?

Reply:
when I look into the log, we panic at the fee contract ownership transfer (as I PM'ed you on Zulip), not light client issue.